src/components/BaseButton.vue

```diff
@@ -0,0 +1,26 @@
+<template>
+  <button
+    :disabled
+    :class="buttonClass"
+    @click="$emit('clickEvent')"
+  >
+    {{ label }}
+  </button>
+</template>
+<script setup lang="ts">
+interface Props {
+  label?: string
+  disabled?: boolean
+  buttonClass?: string
+}
+
+withDefaults(defineProps<Props>(), {
+  label: 'BaseButton',
+  disabled: false,
```

Comment:
You don't need to define a bool prop as false, beacuse if you don't add it in the parent it will be undefined, which is false :)